rp2040-hal/src/spi.rs

```diff
@@ -167,19 +167,30 @@ impl<D: SpiDevice, const DS: u8> Spi<Disabled, D, DS> {
         });
     }
 
+    /// Set master/slave
+    fn set_slave(&mut self, slave: bool) {
+        if slave {
+            self.device.sspcr1.modify(|_, w| w.ms().set_bit());
+        } else {
+            self.device.sspcr1.modify(|_, w| w.ms().clear_bit());
+        }
+    }
+
     /// Initialize the SPI
     pub fn init<F: Into<HertzU32>, B: Into<HertzU32>>(
         mut self,
         resets: &mut RESETS,
         peri_frequency: F,
         baudrate: B,
         mode: &Mode,
+        slave: bool,
```

Comment:
I wonder if it would be better to have a separate function to init the device in slave mode. For example, is the `baudrate`  parameter needed / meaningful if the clock pin is an input?

Also, boolean parameters are often problematic as it may not be obvious what they mean when reading the code. And having a new, separate init method would avoid a breaking change for existing callers.

Reply:
Yeah, good callouts.  I wasn't fond of the parameter, but figured it would be a good starting point for discussion since I'm not familiar with the style of codebase.  A new init method sounds great.

The difference between the two init methods would be pretty slim.  Is there preferred way to reduce duplication since setting baudrate and slave mode happens after the resets and before enabling spi?

Reply:
>  Is there preferred way to reduce duplication since setting the baudrate and slave mode happens after the resets and before enabling spi?

I don't think we have defined a "best practice". For this function, I see at least two viable options:
- Don't worry, just duplicate the code. The method is too small to really matter.
- Have some internal function which is identical with what the code currently does, and make the public functions wrappers for that one. So the documented API is nice and internal code avoids duplication.

Reply:
Sorry for the late response.  I went with the second option, but am happy to move to the first if that works better.